tests/phpunit/MigrateTest.php

```diff
@@ -104,36 +109,129 @@ public function testMigrateSuccess(
+        /** @var JobRunner $destJobRunnerMock */
         $migrate = new Migrate(
-            $sourceClientMock,
-            $destClientMock,
-            $sourceProjectUrl,
-            $sourceProjectToken,
-            $migrateDataOfTablesDirectly,
-            new NullLogger(),
-            $migrateSecrets,
+            $config,
+            $sourceJobRunnerMock,
+            $destJobRunnerMock,
+            'xxx-b',
+            'yyy-b',
+            $loggerMock,
         );
+
+        $sourceClientMock = $this->createMock(StorageClient::class);
+        $sourceClientMock
+            ->method('apiGet')
+            ->willReturnMap([
+                [
+                    'dev-branches/', null, [],
+                    [
+                        [
+                            'id' => '123',
+                            'name' => 'default',
+                            'isDefault' => true,
+                        ],
+                    ],
+                ],
+                [
+                    'components?include=', null, [],
+                    [
+                        [
+                            'id' => 'gooddata-writer', // should be skipped
+                        ],
+                        [
+                            'id' => 'some-component',
+                            'configurations' => [
+                                [
+                                    'id' => '101',
+                                ],
+                                [
+                                    'id' => '102',
+                                ],
+                            ],
+                        ],
+                        [
+                            'id' => 'another-component',
+                            'configurations' => [
+                                [
+                                    'id' => '201',
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ])
+        ;
+        $sourceClientMock
+            ->method('getServiceUrl')
+            ->with('encryption')
+            ->willReturn('https://encryption.keboola.com')
+        ;
+
+        $migrationsClientMock = $this->createMock(Migrations::class);
+
+        if ($migrateSecrets) {
+            $migrationsClientMock
+                ->expects(self::exactly(3))
+                ->method('migrateConfiguration')
+                ->willReturnCallback(function (...$args) {
+                    [, $destinationStack, , , $configId] = $args;
+                    return [
+                        'message' => "Configuration with ID '$configId' successfully " .
+                            "migrated to stack '$destinationStack'.",
+                        'data' => [],
+                    ];
+                });
+
+            $loggerMock->expects(self::exactly(3))
+                ->method('debug')
+                ->withConsecutive(
```

Comment:
Je to deprekovaný moc si na to nezvykej.
Zrovna v tomhle případě je to i docela zbytečný - stačí dát do loggeru TestHandler a pak se podívat přes getRecords, co to zalogovalo

Reply:
Super, takové nějaké řešení jsem původně hledal. Nahrazeno. 🙂